frontend/src/app/modules/challenges/challenge-creation/challenge-creation.component.ts

```diff
@@ -73,41 +75,71 @@ export class ChallengeCreationComponent extends BaseComponent implements OnInit
         private tagService: TagService,
         private toastrService: ToastrNotificationsService,
         private router: Router,
+        private activatedRoute: ActivatedRoute,
     ) {
         super();
-        this.steps = this.stepsData.map(s => s.step);
+        this.steps = this.stepsData.map((s) => s.step);
         this.challenge = getNewChallenge();
         this.challengeVersion = getNewChallengeVersion();
     }
 
     ngOnInit(): void {
-        this.getLanguages();
-        this.getTags();
-        this.getChallengeLevels();
+        this.activatedRoute.paramMap.subscribe((params: ParamMap) => {
+            const challengeId = params.get('id')!;
+
+            this.isEditMode = challengeId !== null;
```

Comment:
```suggestion
            const challengeId = +params.get('id');

            this.isEditMode = !isNaN(challengeId);
```

and parsing in 98 line is unnecessary